### PR TITLE
[FEATURE] runtime caching of unserialized array, possibility to retur…

### PR DESCRIPTION
**Pull request by jgreiner**
…n array

- saving unserialized array in static var to avoid multiple unserialisations in runtime
- possibility to return array (dots removed)
- trimming string result
- minimal call <v:variable.extensionConfiguration /> returns whole extension configuration as array

### CODE REVIEW

File: Classes/ViewHelpers/Variable/ExtensionConfigurationViewHelper.php [hunk 1]
```diff
@@ -30,25 +31,39 @@
...
 			$extensionName = $this->controllerContext->getRequest()->getControllerExtensionName();
 			$extensionKey = GeneralUtility::camelCaseToLowerCaseUnderscored($extensionName);
 		}
-
 		if (FALSE === array_key_exists($extensionKey, $GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf'])) {
 			return NULL;
 		}
 
-		$extConf = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf'][$extensionKey]);
+		if (empty(self::$EXT_CONF[$extensionKey])) {
```
Comment (neufeind) [inline Classes/ViewHelpers/Variable/ExtensionConfigurationViewHelper.php]: I'd use isset or array_key_exists instead here

File: Classes/ViewHelpers/Variable/ExtensionConfigurationViewHelper.php [hunk 2]
```diff
@@ -30,25 +31,39 @@
...
+			self::$EXT_CONF[$extensionKey] = GeneralUtility::removeDotsFromTS(unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf'][$extensionKey]));
+		}
+		if (!$name) {
```
Comment (neufeind) [inline Classes/ViewHelpers/Variable/ExtensionConfigurationViewHelper.php]: if (NULL === name) imho.

**Comment by neufeind** [comment]
The optimization would be fine with me. But imho we can't really remove the dots or trim a string (BC-break).

**Comment by NamelessCoder** [comment]
@neufeind This change (minus the dot stripping) looks good to me. Feel free to cherry-pick, amend (and correct commit subject) then push directly!

**Comment by cedricziel** [comment]
Manually resolved the config and implemented a simple cache with b3d089848c598845b0f2c923c5211e482c740e60

Please review again to make sure I didn't do bad things while playing the keyboard monkey.

Oh and thank you @jgreiner for the idea and the first patch! :)

**Comment by cedricziel** [comment]
Ahh. Dang. I was so into PSR2 already, I didn't remember the huge difference. Opened #978 to apply the new psr2 esque code style to the whole library.

**Comment by NamelessCoder** [comment]
Little too quick on this one. Subject is bad and it's based on a commit quite a while back.

And I can't really say I agree with fixing this code style issue by converting the rest of the repository to the incorrect code style ;)

**Comment by cedricziel** [comment]
Hehe. Agreed. I'll leave the other one open, but revert the merge.

**Comment by NamelessCoder** [comment]
For the reworked version, please take care of previous code comments plus an additional one from me: since abbreviations are discouraged, `$res` should be renamed to something like `$result`.

**Pull request merged**